lib/index.js

```diff
@@ -0,0 +1,45 @@
+'use strict';
+
+const Bluebird = require('bluebird');
+const Joi      = require('joi');
+
+const Validator = require('./validator');
+
+const RADIX = 10;
+
+exports.register = (server, options, next) => {
+
+  server.ext('onPreResponse', (request, reply) => {
+
+    const settings = request.route.settings.plugins.totalCount;
+    const credentials = request.auth.credentials;
+    const filter = request.query.filter || {};
```

Comment:
I feel like we should omit the `|| {}` defaulting. IMO, it should be the responsibility of the `filter` function on the model for handling potentially invalid input and not this plugin, especially considering that all it is doing it is passing it along.